pkg/apiserver/webhook/cloudentityselector_webhook.go

```diff
@@ -98,142 +106,196 @@ func (r *CloudEntitySelector) Default() {
+func (v *CESValidator) validateDelete(req admission.Request) admission.Response { //nolint:unparam
 	// TODO(user): fill in your validation logic upon object deletion.
-	return nil
+	return admission.Allowed("")
 }
 
-func (r *CloudEntitySelector) GetOwnerAccount() (*CloudProviderAccount, error) {
+// GetOwnerAccount fetches the CPA account owning the current CES.
+func (v *CESValidator) GetOwnerAccount(selector *v1alpha1.CloudEntitySelector) (*v1alpha1.CloudProviderAccount, error) {
 	accountNameSpacedName := &types.NamespacedName{
-		Namespace: r.Namespace,
-		Name:      r.Spec.AccountName,
+		Namespace: selector.Namespace,
+		Name:      selector.Spec.AccountName,
 	}
-	ownerAccount := &CloudProviderAccount{}
-	err := client.Get(context.TODO(), *accountNameSpacedName, ownerAccount)
+	ownerAccount := &v1alpha1.CloudProviderAccount{}
+	err := v.Client.Get(context.TODO(), *accountNameSpacedName, ownerAccount)
 	if err != nil {
-		cloudentityselectorlog.Error(err, "failed to find owner account", "account", *accountNameSpacedName)
+		v.Log.Error(err, "failed to find owner account", "account", *accountNameSpacedName)
 		return nil, err
 	}
 	return ownerAccount, err
 }
 
-// validateMatchSections checks for unsupported match combinations and errors out
-func (r *CloudEntitySelector) validateMatchSections() error {
-	// MatchID and MatchName are not supported together in an EntityMatch, applicable for both vpcMatch, vmMatch section
-	for _, m := range r.Spec.VMSelector {
+// validateMatchSections checks for unsupported selector match combinations and errors out.
+func (v *CESValidator) validateMatchSections(selector *v1alpha1.CloudEntitySelector) error {
+	// MatchID and MatchName are not supported together in an EntityMatch, applicable for both vpcMatch, vmMatch section.
+	for _, m := range selector.Spec.VMSelector {
 		if m.VpcMatch != nil {
 			if len(strings.TrimSpace(m.VpcMatch.MatchID)) != 0 &&
 				len(strings.TrimSpace(m.VpcMatch.MatchName)) != 0 {
-				return fmt.Errorf("matchID and matchName are not supported together, " +
-					"configure either matchID or matchName in an EntityMatch")
+				return fmt.Errorf("%s", errorMsgMatchIDNameTogether)
 			}
 		}
 		for _, vmMatch := range m.VMMatch {
 			if len(strings.TrimSpace(vmMatch.MatchID)) != 0 &&
 				len(strings.TrimSpace(vmMatch.MatchName)) != 0 {
-				return fmt.Errorf("matchID and matchName are not supported together, " +
-					"configure either matchID or matchName in an EntityMatch")
+				return fmt.Errorf("%s", errorMsgMatchIDNameTogether)
 			}
 		}
 	}
 
-	ownerAccount, err := r.GetOwnerAccount()
+	ownerAccount, err := v.GetOwnerAccount(selector)
 	if err != nil {
 		return err
 	}
-	cloudProviderType, err := ownerAccount.GetAccountProviderType()
+	cloudProviderType, err := utils.GetAccountProviderType(ownerAccount)
 	if err != nil {
-		cloudentityselectorlog.Error(err, "Invalid cloud provider type")
+		v.Log.Error(err, "Invalid cloud provider type")
```

Comment:
Lets define constant for this error message as well, as it is being used multiple times.

Reply:
done